docs/upgrading.asciidoc

```diff
@@ -0,0 +1,28 @@
+[[upgrading]]
+== Upgrading
+Upgrades between minor versions of the agent, like from 3.1 to 3.2 are always backwards compatible.
+Upgrades that involve a major version bump often come with some backwards incompatible changes.
+
+Before upgrading the agent, be sure to review the:
+
+* <<release-notes,Agent release notes>>
+* {apm-overview-ref-v}/agent-server-compatibility.html[Agent and Server compatibility chart]
+
+[float]
+[[end-of-life-dates]]
+=== End of life dates
+
+We love all our products, but sometimes we must say goodbye to a release so that we can continue moving
+forward on future development and innovation.
+Our https://www.elastic.co/support/eol[End of life policy] defines how long a given release is considered supported,
+as well as how long a release is considered still in active development or maintenance.
+The table below is a simplified description of this policy.
+
+[options="header"]
+|====
+|Agent version |EOL Date |Maintained until
```

Comment:
What do you think about explaining `Maintained until` a bit? Or changing it to something self-explanatory e.g. `feature updates until`, I'm not sure if that's any better though.

Reply:
Since we're following the Stack EOL policy, I'm hesitant to explain it more on this page incase the stack policy changes. The link above (https://www.elastic.co/support/eol[End of life policy]) has all of the information, and we state that "The table below is a simplified description of this policy.".